pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -1050,6 +1050,30 @@ func (c *K8sOrchestrator) IsFSSEnabled(ctx context.Context, featureName string)
 			"Setting the feature state to false", featureName, c.internalFSS.configMapName)
 		return false
 	} else if c.clusterFlavor == cnstypes.CnsClusterFlavorWorkload {
+		// Check if it is WCP defined feature state.
+		if _, exists := common.WCPFeatureStates[featureName]; exists {
+			log.Infof("Feature %q is a WCP defined feature state. Reading the %q configmap in %q namespace.",
+				featureName, common.WCPFSSConfigMapName, common.KubeSystemNamespace)
+			// Check the `wcp-cluster-capabilities` configmap in supervisor for the FSS value.
+			wcpFssMap, err := c.k8sClient.CoreV1().ConfigMaps(common.KubeSystemNamespace).Get(ctx,
+				common.WCPFSSConfigMapName, metav1.GetOptions{})
+			if err != nil {
+				log.Errorf("failed to fetch WCP FSS configmap %q/%q. Setting the feature state "+
+					"to false. Error: %+v", common.KubeSystemNamespace, common.WCPFSSConfigMapName, err)
+				return false
+			}
+			if fssVal, exists := wcpFssMap.Data[featureName]; exists {
+				supervisorFeatureState, err = strconv.ParseBool(fssVal)
+				if err != nil {
+					log.Errorf("Error while converting %q feature state with value: %q in "+
+						"%q/%q configmap to boolean. Setting the feature state to false. Error: %+v", featureName,
+						fssVal, common.KubeSystemNamespace, common.WCPFSSConfigMapName, err)
+					return false
+				}
+				return supervisorFeatureState
+			}
+		}
```

Comment:
Do we need an else {} here to return false ?

Reply:
If the feature state is not present in `wcpFssMap`, then we should check our own CSI configmap, so we do not need an `else` here

Reply:
Alright!

Nit: May be a log message will be useful, to indicate that fss is not in WCP FSS.

Reply:
Most of the FSS won't be available in WCP configmap, that is why I have Info logged the ones we found in the WCP configmap. For all other features, we can assume we are checking the CSI configmap